Assets/HoloToolkit/UX/Scripts/Dialog/DialogButton.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity.Buttons;
+using HoloToolkit.UX.Dialog;
+using UnityEngine;
+
+namespace HoloToolkit.UX.Buttons
+{
+    /// <summary>
+    /// Handling click event and dismiss dialog
+    /// </summary>
+    public class DialogButton : MonoBehaviour
+    {
+        private DialogShell _parentDialog;
+        private Button buttonComponent;
+
+        public DialogShell ParentDialog
+        {
+            get
+            {
+                return _parentDialog;
+            }
+            set
+            {
+                _parentDialog = value;
+            }
+        }
+
+        public Dialog.Dialog.ButtonTypeEnum ButtonTypeEnum;
```

Comment:
This should get easier to read (less awkward) after ButtonTypeEnum is moved out of the class and renamed.